hoot-core-test/src/test/cpp/hoot/core/elements/OsmMapTest.cpp

```diff
@@ -208,10 +211,89 @@ class OsmMapTest : public HootTestFixture
     OsmMapPtr mapB(new OsmMap());
     reader.read("test-files/ToyTestB.osm", mapB);
 
-    NodePtr duplicateNode(
-      new Node(Status::Unknown1, -1669765, 38.85423271202119, -104.89831096020139, 15.0));
-    mapB->addNode(duplicateNode);
-    QString exceptionMsg;
+    const size_t sizeMapAPlusMapBBefore = mapA->getElementCount() + mapB->getElementCount();
```

Comment:
The new behavior for `OsmMap::append` is illustrated in these test changes. The previous version would always fail if the map being appended had an element with the same ID as one in the map being appended to. The new version will compare elements with the same ID, and if they are the same as reported by `ElementComparer`, skip appending the element in the append from map. I also added an option which let's you skip appending them even if they aren't the same.